arch/posix/CMakeLists.txt

```diff
@@ -58,6 +58,9 @@ else() # Linux.x86_64
     check_set_compiler_property(APPEND PROPERTY fpsse2 "SHELL:-msse2 -mfpmath=sse")
     zephyr_compile_options($<TARGET_PROPERTY:compiler,fpsse2>)
     target_compile_options(native_simulator INTERFACE "$<TARGET_PROPERTY:compiler,fpsse2>")
+
+    # Enable 64-bit time_t and file offset values
+    zephyr_compile_definitions(__TIME_BITS=64 __FILE_OFFSET_BITS=64)
```

Comment:
With glibc 2.35 (Ubuntu's 22.04 default) this does not seem to work. With this version it seems there is only 1 underscore for these:
features-time64.h:
``` 
#if defined _TIME_BITS
# if _TIME_BITS == 64
#  if ! defined (_FILE_OFFSET_BITS) || _FILE_OFFSET_BITS != 64
#   error "_TIME_BITS=64 is allowed only with _FILE_OFFSET_BITS=64"
#  elif __TIMESIZE == 32
#   define __USE_TIME_BITS64	1
#  endif
```

Reply:
Yeah, not sure where I came up with the double-underscore versions. I just reviewed current glibc source and it's got the single underscore versions too.